docs/configuration.md

```diff
@@ -194,6 +194,31 @@ syslog {
   # This is the name of the syslog facility to log to.
   facility = "LOCAL5"
 }
+
+# This block defines the configuration for logging to file
+log_file {
+  # If a path is specified, the feature is enabled
+  # Please refer to the documentation for the -log-file
+  # CLI flag for more information about its behaviour
+  path = "/var/log/something.log"
+
+  # This allow you to control the number of bytes that
+  # should be written to a log before it needs to be
+  # rotated. Unless specified, there is no limit to the
+  # number of bytes that can be written to a log file
+  log_rotate_bytes = 1024000
+
+  # This lets you control time based rotation, by default
+  # logs are rotated every 24h
+  log_rotate_duration = "3h"
+
+  # This lets you control the maximum number of older log
+  # file archives to keep. Defaults to 0 (no files are ever
+  # deleted).
+  # Set to -1 to discard old log files when a new one is
+  # created
```

Comment:
I'm inclined to have 0 mean discard old logs (keep 0 logs) and -1 mean keep all logs. Would you mind flipping them or, alternatively, expanding a bit on why you went the way you did?

Reply:
I went this way following the "Tao" for consistency since it looks like it's the way Consul handles this:

![image](https://user-images.githubusercontent.com/29378614/143241281-961e1b5f-2725-4a0e-8d3a-b50590ee7d5c.png)

https://www.consul.io/docs/agent/options#_log_rotate_max_files

Reply:
Apart from consistency, I think I see the reason behind your inclination but the default `0 => never delete` seems saner to me because ultimately logs can be valuable, especially if the user turned on this feature which is off by default.

`-log-file` doesn't exactly scream "log today's stuff to file and forget about the past"

For the same reason `-1 => discard old logs` from a UX perspective sounds good to me because it has to be a user's deliberate choice, probably relevant after they realised that they don't need to keep more than 1 file. 
Useful when logs come in handy for troubleshooting a process that dies suddenly. In these cases you would normally look at the most recent log, yesterday's log is probably not relevant.

For everything else, like for example when file logs are ingested by other tools (that would probably clean after themselves and use the filesystem like a queue), I feel like it's best to have a default that is not destructive. 

I am sure I am missing something but these are my 2cents :)

Happy to hear your thoughts and to change as you see fit.

Reply:
You had me at consistency.

As for the default being safer, I don't think that cuts both ways. During my time as an SRE we always problems with logs filling up the disk on the servers. I can't remember how many log rotation setups I had to tweak. But no matter, I think it is a good idea keeping it consistent with Consul so  no changes needed. Thanks for the engagement!

Reply:
> As for the default being safer, I don't think that cuts both ways.

Not "don't"... should have read "[..] I think that cuts both ways."

Reply:
> always problems with logs filling up the disk on the servers

I hear you, I think that the persistent memory industry is very grateful for all these undeleted logs! :)

> I think it is a good idea keeping it consistent with Consul so no changes needed. Thanks for the engagement!

Awesome, you are very welcome. Thank you too!